extensions/wikia/UserProfilePageV3/maintenance/migrateAvatarsViaService.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+/**
+ * Script migrates the user avatars from DFS to user avatars service
+ *
+ * @see PLATFORM-1419
+ *
+ * @author Macbre
+ * @ingroup Maintenance
+ */
+
+putenv( 'SERVER_ID=177' ); // run in the context of a community wiki
+
+require_once( __DIR__ . '/../../../../maintenance/Maintenance.php' );
+
+class AvatarsMigratorException extends Exception {}
+
+/**
+ * Maintenance script class
+ */
+class AvatarsMigrator extends Maintenance {
+
+	public function execute() {
+		$this->output( "Getting the list of all accounts...\n" );
+
+		// get all accounts
+		global $wgExternalSharedDB;
+		$db = $this->getDB( DB_SLAVE, [], $wgExternalSharedDB );
+
+		$res = $db->select(
+			'`user`',
+			'user_id AS id',
+			[],
+			__METHOD__
+		);
+
+		// process users
+		foreach ( $res as $row ) {
+			try {
+				$this->processUser( User::newFromId( $row->id ) );
+			}
+			catch ( AvatarsMigratorException $e ) {
+				\Wikia\Logger\WikiaLogger::instance()->error( __CLASS__, [
+					'exception' => $e,
+					'user_id'   => $user->getId(),
+				] );
+
+				$this->output( sprintf( "\n\t%s", $e->getMessage() ) );
+			}
+		}
+
+		$this->output( "Processing {$res->numRows()} users...\n" );
+	}
+
+	/**
+	 * Migrate the avatar of a given user
+	 *
+	 * @param User $user
+	 * @throws AvatarsMigratorException
+	 */
+	private function processUser( User $user ) {
+		$this->output( sprintf( "\n[User #%d]: ", $user->getId() ) );
+
+		$avatar = $user->getGlobalAttribute( AVATAR_USER_OPTION_NAME );
+
+		// no avatar set, skip this account
+		if ( is_null( $avatar ) ) {
+			$this->output( 'no avatar set - skipping' );
+			return;
+		}
+
+		// no need to store default avatars in user properties - remove the entry
+		else if ( self::isDefaultAvatar( $avatar ) ) {
+			$this->output( 'default avatar set - removing an attribute' );
+			$user->removeGlobalPreference( AVATAR_USER_OPTION_NAME );
```

Comment:
Oof, sorry for not responding to this earlier...I remember looking at this and thought I had, but clearly didn't.

I don't think we should actually call `$user->removeGlobalPreference( AVATAR_USER_OPTION_NAME );` here. The current behavior when a user deletes their attribute from their profile page is to set the value as an empty string. This is the behavior that I know the discussion service expects from the attributes service (ie, when a user doesn't have an avatar to receive an empty string), and until we update the behavior of deleting avatars from the profile page, we'll end up with a mixture of no values in the db for user's avatars, and empty strings.

So for now I'd argue we should leave them as empty string. They should probably be removed at some point (since I agree, it is silly to store 142k rows of nothing) but we should make sure to coordinate the change with updating the user profile page, and what clients of the attribute service can expect when a user doesn't have an avatar set.